### PR TITLE
Add go context to client http request #70

### PR DESCRIPTION
**Pull request by u5surf**
#70 
If zendesk api server has to return high latency response, It would come to block the other procedure.
So, I added go WithTimeoutContext to zendesk client method using http.request.

### CODE REVIEW

File: zendesk/zendesk.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package zendesk
 
 import (
 	"bytes"
+  "context"
```
Comment (nukosuke) [inline zendesk/zendesk.go]: Please `go fmt` again

**Comment by nukosuke** [comment]
@u5surf 

Thank you for your contribution.

We should have tests for the situation.
Do you have any idea for it?

**Comment by tamccall** [comment]
can we not pass in the client from upstream?

**Comment by nukosuke** [comment]
I agree with @tamccall.

But one thing. Changing func signatures is unacceptable for the next release because I don’t think this high priority issue. This package is already used for upper application and the breaking changes would impact it. 

Instead, Why don’t we add new funcs as alternative to support context?

**Comment by tamccall** [comment]
could go with the approach that the amazon sdk did and try and give everything a with context version. https://aws.amazon.com/blogs/developer/context-pattern-added-to-the-aws-sdk-for-go/

**Comment by nukosuke** [comment]
I marked this long term issue

**Comment by nukosuke** [comment]
the work for context support has been taken over. thanks a lot.

**Pull request closed without merging**